encodings/byte-bool/src/compute/mod.rs

```diff
@@ -135,11 +135,19 @@ impl CompareFn for ByteBoolArray {
 
 impl FillForwardFn for ByteBoolArray {
     fn fill_forward(&self) -> VortexResult<Array> {
-        if self.dtype().nullability() == Nullability::NonNullable {
+        let validity = self.logical_validity();
+        if self.dtype().nullability() == Nullability::NonNullable
+            || validity.all_valid()
+            || validity.all_invalid()
+        {
             return Ok(self.clone().into());
         }
 
-        let validity = self.logical_validity().to_null_buffer()?.unwrap();
+        let validity = self
+            .logical_validity()
+            .to_null_buffer()?
+            .ok_or_else(|| vortex_err!("Failed to convert array validity to null buffer"))?;
```

Comment:
The None here indicates that there's no null buffer, i.e. it's AllValid

Reply:
yep, but I already checked above if it's all_valid() or all_invalid(), so I know that it must be present